Sofa/GUI/Component/src/sofa/gui/component/performer/ConstraintAttachBodyPerformer.inl

```diff
@@ -89,21 +90,19 @@ bool ConstraintAttachBodyPerformer<DataTypes>::startPartial(const BodyPicked& pi
     m_mstate1 = dynamic_cast<MouseContainer*>(this->m_interactor->getMouseContainer());
     m_mstate2 = mstateCollision;
 
-    type::Vec3d point1;
-    type::Vec3d point2;
-
     using sofa::component::constraint::lagrangian::model::BilateralLagrangianConstraint;
 
 
-
-    this->m_interactionObject = sofa::core::objectmodel::New<BilateralLagrangianConstraint<sofa::defaulttype::Vec3Types> >(m_mstate1, m_mstate2);
-    auto* bconstraint = dynamic_cast< BilateralLagrangianConstraint< sofa::defaulttype::Vec3Types >* >(this->m_interactionObject.get());
+    this->m_interactionObject = sofa::core::objectmodel::New<BilateralLagrangianConstraint<DataTypes> >(m_mstate1, m_mstate2);
+    auto* bconstraint = dynamic_cast< BilateralLagrangianConstraint< DataTypes >* >(this->m_interactionObject.get());
 
     bconstraint->setName("Constraint-Mouse-Contact");
 
-    type::Vec3d normal = point1-point2;
+    static const typename DataTypes::Coord point1 {};
+    static const typename DataTypes::Coord point2 {};
+    static const typename DataTypes::Deriv normal {};
 
-    bconstraint->addContact(normal, point1, point2, normal.norm(), 0, index, point2, point1);
+    bconstraint->addContact(normal, point1, point2, 0.0, 0, index, point2, point1);
```

Comment:
The fact that a contact is added between two null vectors is very confusing. It appears that it was not the case initially: https://github.com/sofa-framework/sofa/commit/b664374cd88a3b3db25ab6c50e535b52fd774886. It worth having a discussion on this during the next meeting

Reply:
The fact that a contact was added in the first time (null or not)  is strange, the bilateral interaction should be sufficient.

Reply:
Looking at the code, the only parameters of this function used are the two coords (`point1` and `point2`) and the two indices (`0` and `index`) the rest are just unused and thus not necessary (for the templated method and the rigid spec).